Extension/src/LanguageServer/ui.ts

```diff
@@ -426,14 +435,15 @@ export class LanguageStatusUI {
     private showConfigureIntelliSenseButton: boolean = false;
     private configureIntelliSenseTimeout?: NodeJS.Timeout;
 
-    public async ShowConfigureIntelliSenseButton(show: boolean, client?: Client): Promise<void> {
+    public async ShowConfigureIntelliSenseButton(show: boolean, client: Client, configurationType?: ConfigurationType, sender?: string): Promise<void> {
+        if (configurationType !== undefined && sender !== undefined && this.showConfigureIntelliSenseButton && !show) { // Only log telemetry when the button is being hidden.
+            telemetry.logLanguageServerEvent('showConfigureIntelliSenseButton', { configurationType, sender });
+        }
         if (!await telemetry.showStatusBarIntelliSenseButton() || client !== this.currentClient) {
```

Comment:
Should this `telemetry.showStatusBarIntelliSenseButton` check occur first?  Or, if the intention is to log this telemetry even if that particular feature is disabled (experiment is not active for the user), I see at least one place where `ShowConfigureIntelliSenseButton` will not get called if the feature is not enabled. (In `handleConfigStatusOrPrompt`, if `statusBarIndicatorEnabled` is false).  I'm guessing we'll want to make sure all the calls to this function get through in order to provide telemetry, even when that experiment is not enabled?

Reply:
I don't understand, why would we want to make calls to the function if the experiment is not enabled?